### DISCUSSION TITLE
Add support to MPP Solar

### DISCUSSION THREAD

**Discussion by 74mzsolt** [slipx06/sunsynk-power-flow-card#369] [MPP Solar]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

some functionality of the MPP Solar is not working

### Expected behaviour

MPP Solar functions are more supported

### Mode

lite

### Context / Reason

...

**Comment by Roving-Ronin** [comment]
@74mzsolt Giving a bit more information would assist the developer. I assume your referring to: https://www.mppsolar.com/v3/

If so you also need to provide him with a link/photo of these inverters (or a model that identifies the brand for the majority of users, not just your own model), as well as also providing him with all the battery and inverter status / alert messages that you inverter may output.

e.g: https://github.com/slipx06/sunsynk-power-flow-card/blob/master/src/inverters/brands/huawei.ts

However this will also depend upon the integration that your using to make the inverter available/visible within HA to start with.   Do you have this already setup and able to read the inverter+battery data?

Side Note: Looking at this inverter and wondering is it just another PIP inverter ?

**Comment by 74mzsolt** [comment]
hi, thanks for your reply.

This mainly good for many of us, MPP Solar users - (https://www.thunor.eu/en/products/mpp-solar-pip5048mgx) 
Yes, I have a setup and I can read battery data as well from Pylontech Batteries

**Comment by Roving-Ronin** [comment]
@74mzsolt as mentioned before, to ask slipx06 to add an inverter to the card, so at a minimum a image of the inverter appears, required you provide a high quality image of the inverter. 

YOU need to do that, I've tried looking myself and this inverter maker doesn't appear to have any, with all their images that are higher quality all having the inverters appearing at an angle (instead of front on).

If you the inverter doesn't work with the card and the expected sensors readings (i.e. card expects say Status = Normal, but your sensors reply with 'OK', then you also need to provide all the status / alarms / conditions etc that your inverter reports to allow them to be coded into the card).  Also linked to this is HOW your accessing the data on the inverter and pulling it into Home Assistant ? As mentioned before, what integration are you using to do that ? You need to be using a documented integration (or setup a GitHub Repo) and create a (good) solution for doing this, as there is no use adding an inverter unless there is a common way for everyone to integrate it into their Home Assistant. If there's isn't a common way for everyone to do that for a particular brand of inverter, the card will only be able to provide the inverter image and no one else will have the correct status / alarms / conditions etc that your inverter reports that then works with the card.

Slipx06 makes a great card and is extremely responsive to requests, but you need to provide him with the required information if he is to add any inverter to the card.

**Comment by slipx06** [comment]
Thanks @Roving-Ronin.
@74mzsolt I've managed to find an image of your inverter. I'll add it in the next release.

**Comment by 74mzsolt** [comment]
You are amazing, thanks a lot

**Comment by slipx06** [comment]
I'm moving this to a discussion. If you share the inverter status codes I can add those.

**Comment by slipx06** [comment]
@74mzsolt  the image has been added to https://github.com/slipx06/sunsynk-power-flow-card/releases/tag/v4.29.0

**Comment by 74mzsolt** [comment]
![error](https://github.com/slipx06/sunsynk-power-flow-card/assets/99143738/5b352ed2-547d-4212-bc7d-1b7314e71235)

**Comment by 74mzsolt** [comment]
![warning](https://github.com/slipx06/sunsynk-power-flow-card/assets/99143738/cfe8a2cd-74ef-4d60-a1c7-1e870eb22e73)

**Comment by slipx06** [comment]
What integration method are you using and can you share the sensor that reports the inverter status

**Reply by 74mzsolt** [reply]
the sensor data is from mqqt dataflow, and it is called: "sensor.device_mode" ,with more inverters in parallel "sensor.inverter_2_device_mode", "sensor.inverter_3_device_mode" etc.

they result in "Solar/Battery", or "Grid" or "Error X"

**Reply by slipx06** [reply]
Please share some sample sensor output from your HA. I also need a link to the integration i.e. Github etc

**Comment by 74mzsolt** [comment]
sensor.battery_power
sensor.battery_state_of_charge
sensor.grid_power
sensor.inverter_load_power
select.inverter_operation_logic
sensor.inverter_pv_power
sensor.inverter_pv_current
sensor.inverter_temperature

**Comment by 74mzsolt** [comment]
sensor.load_power
